WalletWasabi.Gui/Controls/WalletExplorer/HistoryTabViewModel.cs

```diff
@@ -125,20 +125,19 @@ private static List<(DateTimeOffset dateTime, Height height, Money amount, strin
 			{
 				var found = txRecordList.FirstOrDefault(x => x.transactionId == coin.TransactionId);
 
-				if (Global.WalletService is null) // disposed meanwhile
-				{
-					break;
-				}
+				if (Global.WalletService is null) break; // Disposed meanwhile.
 
-				SmartTransaction foundTransaction = Global.WalletService.TransactionCache.FirstOrDefault(x => x.GetHash() == coin.TransactionId);
+				SmartTransaction foundTransaction = Global.WalletService?.TransactionCache?.FirstOrDefault(x => x.GetHash() == coin.TransactionId);
 				if (foundTransaction is null)
 				{
+					if (Global.WalletService is null) break; // Disposed meanwhile.
 					continue;
 				}
 
 				DateTimeOffset dateTime;
 				if (foundTransaction.Height.Type == HeightType.Chain)
 				{
+					if (Global.WalletService is null) break; //  Disposed meanwhile. Actually caught an NRE here.
```

Comment:
This kind of code is useless, because nothing will prevent `Global.WalletService` to become null between this call and the next line. In which case, it will throw NRE.

The way you should do it is by capturing `Global.WalletService` in a local variable at the beginning of the method.

Reply:
Yeah, that's pretty legacy code there. Opened an issue that'll remove this code: https://github.com/zkSNACKs/WalletWasabi/issues/1292

Reply:
Yeah, or just inject the walletservice instead of depending on global.